### PR TITLE
Fix docstring in AdamW

### PR DESCRIPTION
**Pull request by craffel**
The definition of AdamW in the `torch.optim.AdamW` docstring currently sets $$\theta_t \leftarrow \theta_t \ldots$$ in two places. You can't use $$\theta_t$$ to update $$\theta_t$$, so this PR fixes the subscript on the right to be $$t - 1$$. See the Adam docstring for comparison: https://pytorch.org/docs/stable/generated/torch.optim.Adam.html

### CODE REVIEW

**Comment by janeyx99** [comment]
Ah, this is actually intentional, because decoupled weight decay created a theta_t and these update should use that updated result.

So it'd be incorrect if we switch to theta_{t-1}

**Comment by craffel** [comment]
Ah, I see what you mean! To be overly pedantic, I would still suggest not ever writing $\theta_t \leftarrow \theta_t ...$ - I would instead say that the weight decay line should be $\theta_t^\prime \leftarrow \theta_{t-1} - \gamma \lambda \theta_{t-1}$ or something, and then $\theta_t^\prime$ replaces $\theta_t$ thereafter. FWIW the AdamW paper avoids this via some different conventions on the statement of the algorithm. Of course, if you don't care about being pedantic, you can just close this PR!

**Comment by janeyx99** [comment]
lol if u want to change this pr to do that, i’d accept it
